src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -877,6 +888,7 @@ CATCH_LOG()
 // - isVisible: whether the cursor should be visible
 void Terminal::SetCursorOn(const bool isOn) noexcept
 {
+    auto lock = LockForWriting();

```

Comment:
Do we need an unlock somewhere?

Reply:
Nope! It'll release when destructed (goes out of scope).